### PR TITLE
Enhance ring buffer implementation with additional features and tests

### PR DESCRIPTION
**Pull request by bugparty**
- Updated CMakeLists.txt to define project and include Google Test conditionally.
- Expanded .gitignore to include build artifacts and temporary testing files.
- Improved RingBuffer class with move semantics, swap functionality, and additional iterator methods.
- Added comprehensive unit tests for copy behavior, iterator equality, and buffer state after operations.
- Updated README with FetchContent integration instructions and testing policy.

----

> [!IMPORTANT]
> Enhance `RingBuffer` with move semantics, swap functionality, iterator improvements, and add comprehensive tests.
> 
>   - **RingBuffer Enhancements**:
>     - Added move semantics and swap functionality to `ring_buffer` in `RingBuffer.hpp`.
>     - Enhanced iterator methods in `ring_buffer_iterator` in `RingBuffer.hpp`.
>   - **CMake and Build**:
>     - Updated `CMakeLists.txt` to conditionally include Google Test based on `RINGBUFFER_BUILD_TESTS` option.
>     - Added `RingBuffer` as an interface library in `CMakeLists.txt`.
>   - **Documentation**:
>     - Updated `README.md` with FetchContent integration instructions and testing policy.
>   - **Testing**:
>     - Added unit tests in `test_main.cpp` for move semantics, swap functionality, iterator equality, and buffer state after operations.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=bugparty%2FRingBufferCpp&utm_source=github&utm_medium=referral)<sup> for 002c14f8d36ad509634445e606f84518700bd2bb. You can [customize](https://app.ellipsis.dev/bugparty/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

## Summary by CodeRabbit

* **New Features**
  * Added move/swap support, element accessors (front/back/index), pop_front, clear, public iterators, and state queries (size/capacity/empty/full)

* **Documentation**
  * Expanded README with updated clone/build instructions, FetchContent integration examples, usage snippets, and a testing policy (how to skip building tests)

* **Tests**
  * Large new test suite covering moves, swaps, iteration, wrap/overwrite, and edge cases; test builds can be toggled via a configuration option

* **Chores**
  * Updated ignore patterns and added repository housekeeping files

### CODE REVIEW

File: RingBuffer.hpp [hunk 1]
```diff
@@ -88,13 +92,13 @@ namespace buffers {
...
                         ring_buffer_iterator<T,N,C,Overwrite> const& r) noexcept {
-            return l.count() == r.count();
+            return l.source() == r.source() && l.count() == r.count() && l.index() == r.index();
         }
 
         template<typename T, size_t N, bool C, bool Overwrite>
         bool operator!=(ring_buffer_iterator<T,N,C,Overwrite> const& l,
                         ring_buffer_iterator<T,N,C,Overwrite> const& r) noexcept {
-            return l.count() != r.count();
+            return l.source() != r.source() || l.count() != r.count() || l.index() != r.index();
```
[the remark this answers is not in the document]
Comment (bugparty) [reply RingBuffer.hpp]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bugparty/RingBufferCpp/pull/2#discussion_r2702272012)

File: test_main.cpp [hunk 2]
```diff
@@ -96,6 +96,246 @@ TEST(RingBufferTest, Test7ArrowOperator) {
...
+
+    using std::swap;
+    swap(b1, b2);
+
+    EXPECT_EQ(b1.size(), 1);
+    EXPECT_EQ(b1.front(), 7);
+    EXPECT_EQ(b2.size(), 2);
+    EXPECT_EQ(b2.front(), 3);
+    EXPECT_EQ(b2.back(), 4);
+}
```
[the remark this answers is not in the document]
Comment (bugparty) [reply test_main.cpp]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bugparty/RingBufferCpp/pull/2#discussion_r2702272021)

File: test_main.cpp [hunk 3]
```diff
@@ -96,6 +96,246 @@ TEST(RingBufferTest, Test7ArrowOperator) {
...
+    std::vector<int> expected = {2, 3, 4, 5};
+    EXPECT_EQ(values, expected);
+}
+
+TEST(RingBufferTest, IteratorsFromDifferentBuffersAreNotEqual) {
+    ring_buffer<int, 3> b1;
+    ring_buffer<int, 3> b2;
+
+    EXPECT_NE(b1.cbegin(), b2.cbegin());
+}
```
[the remark this answers is not in the document]
Comment (bugparty) [reply test_main.cpp]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bugparty/RingBufferCpp/pull/2#discussion_r2702272036)

**Comment by bugparty** [comment]
@codex review

**Comment by bugparty** [comment]
@codex review
